src/plugins/discover/public/application/main/discover_main_route.tsx

```diff
@@ -79,34 +79,32 @@ export function DiscoverMainRoute(props: Props) {
         const hasUserDataViewValue = await data.dataViews.hasData
           .hasUserDataView()
           .catch(() => false);
-
-        const hasESDataValue =
-          isDev || (await data.dataViews.hasData.hasESData().catch(() => false));
         setHasUserDataView(hasUserDataViewValue);
-        setHasESData(hasESDataValue);
```

Comment:
I guess it still makes sense to keep it at the top so we have a calculated `hasESData`  value for the following logic when an earlier return from the function happens:
```
hasESData: () => Promise.resolve(isDev ? true : hasESData),
```